roles/KubevirtTemplateValidator/tasks/main.yml

```diff
@@ -0,0 +1,42 @@
+---
+# tasks file for KubevirtTemplateValidator
+- name: Set template:view role
+  k8s:
+    state: present
+    definition: "{{ lookup('file', 'template-view-role.yaml') }}"
+- name: Make certificates
+  shell: /usr/local/bin/webhook-mkcert.sh
+  register: mkcert
+- name: Parse mkcert output
+  set_fact:
+    certsdata: "{{ mkcert.stdout | from_json }}"
+- name: Send CSR
+  k8s:
+    state: present
+    src: "{{ certsdata.webhookcsr }}"
+# TODO: wait for CSR to be created
+- name: Approve CSR
+  command: /usr/bin/oc adm certificate approve {{ certsdata.csrname }}
+- name: Fetch certificate from CSR
+  set_fact:
+    # TODO: certsdata.csrname doesn't work. Maybe different parameter handling?
+    csrdata: "{{ lookup('k8s', kind='CertificateSigningRequest', api_version='certificates.k8s.io/v1beta1', resource_name='virtualmachine-template-validator.kubevirt') }}"
```

Comment:
You can also use k8s_facts module and register: the output, lookup works too though

Reply:
Seems much nicer, thanks! done